clang/lib/Analysis/LifetimePsetBuilder.cpp

```diff
@@ -633,6 +633,16 @@ class PSetsBuilder : public ConstStmtVisitor<PSetsBuilder, void> {
     if (!Callee)
       return;
 
+    if (auto MCE = dyn_cast<CXXMemberCallExpr>(CallE)) {
```

Comment:
This is good for getting rid of the false positive but in the future I think it would be great to:
* Update the state, so we know what is the pointee of the returned ptr (*uniqueptr in our case).
* Set the unique_ptr to null. I am not sure if we support nullable owners properly at the point. The current paper also does not really elaborate on this.
* Maybe we need a way to annotate ownership transferring functions in the future? Not all of them migh have the name release.

Some of these notes are more for our future selves designing a new version of the analysis, rather than the current implementation :)